src/Utilities/FractionUtilities.hpp

```diff
@@ -69,25 +67,26 @@ class ContinuedFraction {
 
   /// Advance to the next element in the expansion
   ContinuedFraction& operator++() noexcept {
-    if (remainder_ == 0 or (error_ /= square(remainder_)) > 1) {
```

Comment:
Am I right in understanding that before, by squaring the remainder, the expansion stopped too soon, because it was much easier for error_ > square(remainder_) than error_ > remainder_?

Reply:
Correct.

For detail, the next term is roughly `1/remainder_`, so, by the naive error propagation formulas, it has the same relative error as `remainder_`, so absolute error `error_/remainder_^2`.  We stopped when this error exceeded 1, because then we didn't know the value of the next term.

The problem is that we might know that our error bounds exclude `remainder_ = 0`, even if we don't know what the exact value is.  Returning an approximate value within the error bars is then better than dropping the remainder entirely.